feat: rewrite signup and signin page with tailwind UI

olamide203 opened the pull request:
<!-- If your PR fixes an open issue, use `Closes #101` to link your PR with the issue. #101 stands for the issue number you are fixing -->
### Fixes
- a rewrite of the signup and signing page with tailwind UI


Closes #100 

### 📄 Note to reviewers

- used react-icons for social media icons on signup and sign in page


### 📷 Screenshots
![Screenshot 2022-08-20 at 15-54-53 Defaang](https://user-images.githubusercontent.com/66855179/185753114-78c2823d-d627-423b-957e-2b886b4710f1.png)

![Screenshot 2022-08-20 at 15-54-28 Defaang](https://user-images.githubusercontent.com/66855179/185753349-b2f0823a-22b1-4a9e-87b8-a189300a40f9.png)

![Screenshot 2022-08-20 at 16-04-58 Defaang](https://user-images.githubusercontent.com/66855179/185753355-bea39771-a7c5-4295-8139-105afe662edb.png)

ykdojo commented on .env.template:
```diff
@@ -1,2 +0,0 @@
-NEXT_PUBLIC_SUPABASE_URL=YOUR_SUPABASE_URL
-NEXT_PUBLIC_SUPABASE_ANON_KEY=YOUR_SUPABASE_ANON_KEY
```
don't delete this file

olamide203 replied:
sorry, I renamed the original without making a copy first. would make a PR with changes

ykdojo commented on pages/signin.tsx:
```diff
@@ -26,10 +26,39 @@ const SignIn: NextPage = () => {
+							</h2>
+						</div>
+						<div className='mt-6'>
+							<p className='text-sm text-gray-700'>sign in with</p>
```
sign in -> Sign in

ykdojo commented on pages/signin.tsx:
```diff
@@ -38,37 +67,79 @@ const SignIn: NextPage = () => {
+									/>
+									<label htmlFor='remember-me' className='ml-2 block text-sm text-gray-900'>
+										{' '}
+										Remember me{' '}
```
do we have these functionalities yet?

ykdojo replied:
(if not, let's comment them out or delete them)

olamide203 replied:
I don't think that functionality has been implemented yet. i'll comment out that part

ykdojo commented on pages/signup.tsx:
```diff
@@ -39,59 +65,71 @@ const SignUp: NextPage = () => {
-								<p>Confirm Password</p>
+							<div className='flex w-full flex-col space-y-2 '>
+								<label htmlFor='confirm-password' className='text-sm text-gray-600'>
+									Confirm Password
```
what is this for?

olamide203 replied:
just to confirm password before signup. it was in the former signup form so I added it to the new one also

olamide203 replied:
should I remove it?

iShibi replied:
I suggest keeping the `Confirm Password` input as it forces users to double-check their password before signing up. Can you add this back?

ykdojo replied:
@iShibi is there any modern website that uses that field? I just haven't really seen it recently.

ykdojo replied:
Plus as long as the email is correct, the user can always reset their password. So if anything, there should be a "confirm email" field, but I don't think that's necessary, either.

iShibi replied:
That's a valid point. I guess we are good without it.

ykdojo commented on pages/signup.tsx:
```diff
@@ -26,11 +26,37 @@ const SignUp: NextPage = () => {
+							<h2 className='mt-6 text-start text-3xl font-bold tracking-tight text-gray-900'>Create an Account</h2>
+						</div>
+						<div className='mt-6'>
+							<p className='text-sm text-gray-700'>sign up with</p>
```
sign up -> Sign up

ykdojo commented on pages/signup.tsx:
```diff
@@ -26,11 +26,37 @@ const SignUp: NextPage = () => {
+					<div className='h-full max-w-md p-4 lg:h-[650px] lg:w-[450px]'>
+						{/* <h1 className='p-2 text-3xl font-semibold'>Create an Account ⚡</h1> */}
+						<div>
+							<h2 className='mt-6 text-start text-3xl font-bold tracking-tight text-gray-900'>Create an Account</h2>
```
Create an Account -> Create an account

ykdojo commented on pages/signup.tsx:
```diff
@@ -26,11 +26,37 @@ const SignUp: NextPage = () => {
-				<div className='flex h-screen  w-[100%] flex-col items-center justify-center bg-gray-100 lg:w-[50%]'>
+			<main className='flex h-max w-screen'>
+				<div className='flex w-[100%] flex-col items-center justify-center bg-gray-50 lg:w-[40%]'>
 					{/* Now you can decide different login ways - primarily I am giving google and twiiter for more add it yourself. */}
```
delete this line

ykdojo commented on pages/signup.tsx:
```diff
@@ -26,11 +26,37 @@ const SignUp: NextPage = () => {
-					<div className='h-full w-full  bg-white p-4 md:shadow-lg lg:h-[650px] lg:w-[450px]'>
-						<h1 className='p-2 text-3xl font-semibold'>Create an Account ⚡</h1>
+					<div className='h-full max-w-md p-4 lg:h-[650px] lg:w-[450px]'>
+						{/* <h1 className='p-2 text-3xl font-semibold'>Create an Account ⚡</h1> */}
```
delete this line

ykdojo commented on pages/signin.tsx:
```diff
@@ -38,37 +67,79 @@ const SignIn: NextPage = () => {
+						<p className='my-2 flex w-full justify-center text-sm text-gray-700'>
 							Not registered yet?
 							<Link className='cursor-pointer font-bold' href={'/signup'}>
 								<a className=' text-blue-600 hover:underline'> Sign up</a>
```
There should be a space before "Sign up"

ykdojo reviewed (changes requested):
Thank you! It looks much better than what we had before, but I think the registration/signin box should be in a white background.

olamide203 commented:
I've made changes to the areas that were pointed out
also used the image from the code template in place of the former one

###Screenshot
![Screenshot 2022-08-20 at 19-08-42 Defaang](https://user-images.githubusercontent.com/66855179/185760771-acbc5d24-38c6-42b0-8e29-a5ee128e19fc.png)

ykdojo commented:
Thank you. The new UI has a lot of whitespace on the bottom, though:

<img width="1536" alt="image" src="https://user-images.githubusercontent.com/1811651/185762389-55f420df-f7b4-4e27-8920-3734c45299d4.png">

Also, the "sign in with" options should be Google and GitHub - or we can comment them out for now.

rohitdasu commented on pages/signup.tsx:
```diff
@@ -26,84 +25,159 @@ const SignUp: NextPage = () => {
+							Already have an account?{' '}
 							<Link className='cursor-pointer font-bold' href={'/signin'}>
-								<a className=' text-blue-600 hover:underline'> Sign in</a>
+								<a className=' mx-1 text-blue-600 hover:underline'> Sign in</a>
```
please add a space before Sign in

rohitdasu commented on pages/signin.tsx:
```diff
@@ -23,13 +22,74 @@ const SignIn: NextPage = () => {
-					<div className='w-full  bg-white p-4 md:shadow-lg lg:h-[500px] lg:w-[400px]'>
-						<h1 className='p-2 text-3xl font-semibold'>Sign in ⚡</h1>
+					<div className='mx-auto w-full max-w-sm lg:w-96'>
+						{/* <h1 className='p-2 text-3xl font-semibold'>Sign in ⚡</h1> */}
```
please remove it if it's not needed

rohitdasu commented on pages/signin.tsx:
```diff
@@ -23,13 +22,74 @@ const SignIn: NextPage = () => {
+										className='inline-flex w-full justify-center rounded-md border border-gray-300 bg-white py-2 px-4 text-sm font-medium text-gray-500 shadow-sm hover:bg-gray-50'
+									>
+										<span className='sr-only'>Sign in with Twitter</span>
+										<svg className='h-5 w-5' aria-hidden='true' fill='currentColor' viewBox='0 0 20 20'>
+											<path d='M6.29 18.251c7.547 0 11.675-6.253 11.675-11.675 0-.178 0-.355-.012-.53A8.348 8.348 0 0020 3.92a8.19 8.19 0 01-2.357.646 4.118 4.118 0 001.804-2.27 8.224 8.224 0 01-2.605.996 4.107 4.107 0 00-6.993 3.743 11.65 11.65 0 01-8.457-4.287 4.106 4.106 0 001.27 5.477A4.073 4.073 0 01.8 7.713v.052a4.105 4.105 0 003.292 4.022 4.095 4.095 0 01-1.853.07 4.108 4.108 0 003.834 2.85A8.233 8.2 ...
```
let's not directly put the svg code into the tsx.

rohitdasu commented on pages/signin.tsx:
```diff
@@ -23,13 +22,74 @@ const SignIn: NextPage = () => {
+										className='inline-flex w-full justify-center rounded-md border border-gray-300 bg-white py-2 px-4 text-sm font-medium text-gray-500 shadow-sm hover:bg-gray-50'
+									>
+										<span className='sr-only'>Sign in with GitHub</span>
+										<svg className='h-5 w-5' aria-hidden='true' fill='currentColor' viewBox='0 0 20 20'>
+											<path
+												fillRule='evenodd'
+												d='M10 0C4.477 0 0 4.484 0 10.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.0 ...
```
let's not use svg code inside tsx

olamide203 commented:
### Fixes 

- Removed extra space at the bottom of  UI
- Used react-icons for social icons
- Replaced social signup options with google and github

rohitdasu commented on package.json:
```diff
@@ -14,7 +14,8 @@
 		"react": "18.2.0",
-		"react-dom": "18.2.0"
+		"react-dom": "18.2.0",
+		"react-icons": "^4.4.0"
```
@ykdojo @iShibi @subhoghoshX 
are you guys fine with `react-icons` package? IMO i could have downloaded svgs(fontawesome) and place it inside public folder and use it instead of adding a package.

ykdojo replied:
I don't have a strong opinion on it as long as the performance is reasonably fast!

iShibi replied:
Their docs state that tree-shaking works with this library. If that's true then it is fine to keep it as a dependency.

ykdojo commented:
Thank you so much! It looks great.

A few minor things:

1. I feel like the heading ("Sign in to your account") is slightly too thick?

That might be a styling thing - I might make a separate issue for it instead:

<img width="558" alt="image" src="https://user-images.githubusercontent.com/1811651/185766292-83500a57-b57a-48b6-9761-a3f12d397b3a.png">

2. Should we comment out the social logins/signups until we implement them?

olamide203 commented:
- The heading does look a bit wordy. I think just 'Sign in' should be okay.
- we could also comment out the social logins/signups till they are implemented. I think there is an open issue addressing it.
should I make a new PR with these changes?

ykdojo commented:
Yes! Thank you. I think you can just update this PR

ykdojo commented:
One more thing - let's remove the username field from the signup form.
<img width="582" alt="image" src="https://user-images.githubusercontent.com/1811651/185769036-5ed6947d-2ea3-4c16-a087-3ed9b0178d97.png">

Context: #95

olamide203 commented:
what about the confirm password field? it not currently in use. should it be commented out till the feature gets implemented?

ykdojo commented:
Let's actually remove that one. I don't see a lot of websites using that field nowadays, and I don't think it's necessary.

iShibi commented on pages/signin.tsx:
```diff
@@ -38,49 +72,94 @@ const SignIn: NextPage = () => {
+							Not registered yet?{' '}
 							<Link className='cursor-pointer font-bold' href={'/signup'}>
-								<a className=' text-blue-600 hover:underline'> Sign up</a>
+								<a className='mx-1 text-blue-600 hover:underline'> Sign up</a>
```
No need to add space here as you already added a `{' '}` on line 149 after `Not registered yet?`
```suggestion
								<a className='mx-1 text-blue-600 hover:underline'>Sign up</a>
```

iShibi commented on pages/signup.tsx:
```diff
@@ -1,109 +1,124 @@
+									Email address
+								</label>
 								<input
 									type='text'
```
```suggestion
									type='email'
```

iShibi commented on pages/signin.tsx:
```diff
@@ -38,49 +72,94 @@ const SignIn: NextPage = () => {
+									Email address
+								</label>
 								<input
 									type='text'
```
```suggestion
									type='email'
```

iShibi commented:
Looks good to me. We can use some `SignIn` and `SignUp` components here to keep the pages tidy, what do you think?

olamide203 commented:
yea, both forms could be moved to separate components with a `handleSubmit` prop which would get called on submit

iShibi commented:
I think this is ready to merge. In the next iteration, we can add options for usernames and socials. Just run `npm run format:fix` to resolve the prettier error and we are good to go.

benny123tw commented on components/SignUp/Form.tsx:
```diff
@@ -0,0 +1,61 @@
+
+export const SignUpForm = () => {
+	const [email, setEmail] = useState('');
+	const [password, setPassowrd] = useState('');
```
typo `setPassowrd` -> `setPassword`

benny123tw commented on components/SignUp/Form.tsx:
```diff
@@ -0,0 +1,61 @@
+export const SignUpForm = () => {
+	const [email, setEmail] = useState('');
+	const [password, setPassowrd] = useState('');
+	const handleSignup = async () => {
```
maybe using camel case? `handleSignup` -> `handleSignUp`

ykdojo commented:
Since I have to interact with this part of the code as I work on #108, I'm going to merge it as is.

I might raise issues/propose fixes as I go.
